feat: Add duration doc page (#467)
stackabletech/documentation: modules/concepts/pages/duration.adoc

= Duration format

:rust-duration-max: https://doc.rust-lang.org/std/time/struct.Duration.html#associatedconstant.MAX
:go-std-time: https://cs.opensource.google/go/go/+/refs/tags/go1.21.2:src/time/format.go;l=1589
:k8s-cr: https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/
:go: https://go.dev/

> All Stackable operators use a human-readable duration format. It very closely resembles the format used by the {go}[Go] programming language.

Reviewer: ```suggestion
All Stackable operators use a human-readable duration format. It very closely resembles the format used by the {go}[Go] programming language - which Kubernetes uses internally.
```

> Every duration field of a {k8s-cr}[CustomResource], for example the xref:trino:usage-guide/operations/graceful-shutdown.adoc[`spec.clusterConfig.gracefulShutdownTimeout`] field, supports this format.

Reviewer: As discussed this morning :)
```suggestion
Every duration field of a {k8s-cr}[CustomResource], for example the xref:trino:usage-guide/operations/graceful-shutdown.adoc[`spec.workers.roleConfig.gracefulShutdownTimeout`] field, supports this format.
```

There is no official format specification, but the source code of {go-std-time}[`time.ParseDuration`] in the Go standard library can be used as an implementation reference.
The format looks like this:

[source,yaml]
----

> # ...
> spec:
>   clusterConfig:
>     gracefulShutdownTimeout: 15d18h34m42s

Reviewer: I think I would keep the example generic and let the user figure out where to stuff it in. (users are stupid and will copy/paste this snippet otherwise ^^)
```suggestion
15d18h34m42s
```

Author: Changed in ea9126f

= Duration format

:rust-duration-max: https://doc.rust-lang.org/std/time/struct.Duration.html#associatedconstant.MAX
:go-std-time: https://cs.opensource.google/go/go/+/refs/tags/go1.21.2:src/time/format.go;l=1589
:k8s-cr: https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/
:go: https://go.dev/

All Stackable operators use a human-readable duration format. It very closely resembles the format used by the {go}[Go] programming language - which Kubernetes uses internally.
Every duration field of a {k8s-cr}[CustomResource], for example the xref:trino:usage-guide/operations/graceful-shutdown.adoc[`spec.workers.roleConfig.gracefulShutdownTimeout`] field, supports this format.
There is no official format specification, but the source code of {go-std-time}[`time.ParseDuration`] in the Go standard library can be used as an implementation reference.

> The format looks like this: `15d18h34m42s`.

Reviewer: IMHO this improves readability
```suggestion

The format looks like this: `15d18h34m42s`.
```

Author: Changed in 900e98c
